services/device-registry-mongodb/src/test/java/org/eclipse/hono/deviceregistry/mongodb/model/MongoDbBasedDeviceDtoTest.java

```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.deviceregistry.mongodb.model;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+
+import org.eclipse.hono.deviceregistry.mongodb.utils.MongoDbDeviceRegistryUtils;
+import org.eclipse.hono.service.management.BaseDto;
+import org.eclipse.hono.service.management.device.Device;
+import org.eclipse.hono.service.management.device.DeviceStatus;
+import org.eclipse.hono.util.RegistryManagementConstants;
+import org.junit.jupiter.api.Test;
+
+import io.vertx.core.json.JsonObject;
+
+
+class MongoDbBasedDeviceDtoTest {
+
+    /**
+     * Tests that the DTO is serialized to JSON as expected in order to verify that the document persisted is
+     * written as expected.
+     */
+    @Test
+    public void testEncode() {
+        final String tenantId = "barfoo";
+        final String deviceId = "foobar";
+        final Device device = new Device();
+        final DeviceStatus deviceStatus = new DeviceStatus()
+                .setAutoProvisioned(true)
+                .setAutoProvisioningNotificationSent(true)
+                // make sure that these values do not interfere with the corresponding properties of DeviceDto's parent classes
+                .setCreationTime(Instant.now().plusSeconds(3600))
+                .setLastUpdate(Instant.now().plusSeconds(3600));
+        final Instant created = Instant.now().minusSeconds(1).truncatedTo(ChronoUnit.SECONDS);
+        final Instant updated = Instant.now().truncatedTo(ChronoUnit.SECONDS);;
+        final String version = "spam";
+
+        final var deviceDto = MongoDbBasedDeviceDto.forRead(MongoDbBasedDeviceDto::new, tenantId, deviceId, device, deviceStatus,
```

Comment:
The reason I didn't use one of those is that I wanted to test if every property of the DTO is serialised as expected. If I had used forCreate() or forUpdate() not all fields would be populated and the test code would be partially duplicated.

Maybe it would help if I put what I stated above into a comment?

Reply:
FMPOV it would be more helpful to have separate test methods which test the DTO in the way(s) that it is supposed to be used by client code ...